subprojects/testing-jvm/src/integTest/groovy/org/gradle/testing/testsuites/dependencies/TestSuitesKotlinDSLDependenciesIntegrationTest.groovy

```diff
@@ -0,0 +1,2252 @@
+                    }
+                }
+            }
+        }
+
+        tasks.named("check") {
+            dependsOn(testing.suites.named("integTest"))
+        }
+
+        tasks.register("checkConfiguration") {
+            dependsOn("test", "integTest")
+            doLast {
+                val testCompileClasspathFileNames = configurations.getByName("testCompileClasspath").files.map { it.name }
+                val testRuntimeClasspathFileNames = configurations.getByName("testRuntimeClasspath").files.map { it.name }
+
+                assert(testCompileClasspathFileNames.containsAll(listOf("commons-lang3-3.11.jar", "servlet-api-3.0-alpha-1.jar", "guava-30.1.1-jre.jar")))
+                assert(!testCompileClasspathFileNames.contains("mysql-connector-java-8.0.26.jar")) { "runtimeOnly dependency" }
+                assert(testRuntimeClasspathFileNames.containsAll(listOf("commons-lang3-3.11.jar", "guava-30.1.1-jre.jar", "mysql-connector-java-8.0.26.jar")))
+                assert(!testRuntimeClasspathFileNames.contains("servlet-api-3.0-alpha-1.jar")) { "compileOnly dependency" }
+
+                val integTestCompileClasspathFileNames = configurations.getByName("integTestCompileClasspath").files.map { it.name }
+                val integTestRuntimeClasspathFileNames = configurations.getByName("integTestRuntimeClasspath").files.map { it.name }
+
+                assert(integTestCompileClasspathFileNames.containsAll(listOf("servlet-api-2.5.jar", "guava-29.0-jre.jar")))
+                assert(!integTestCompileClasspathFileNames.contains("commons-lang3-3.11.jar")) { "implementation dependency of project, should not leak to integTest" }
+                assert(!integTestCompileClasspathFileNames.contains("mysql-connector-java-6.0.6.jar")) { "runtimeOnly dependency" }
+                assert(integTestRuntimeClasspathFileNames.containsAll(listOf("commons-lang3-3.11.jar", "guava-29.0-jre.jar", "mysql-connector-java-6.0.6.jar")))
+                assert(!integTestRuntimeClasspathFileNames.contains("servlet-api-2.5.jar")) { "compileOnly dependency" }
+            }
+        }
+        """
+
+        expect:
+        succeeds 'checkConfiguration'
+    }
+
+    def 'can add dependencies to the implementation, compileOnly and runtimeOnly configurations of a suite using a GAV map'() {
+        given:
+        buildKotlinFile << """
+        plugins {
+          `java-library`
+        }
+
+        ${mavenCentralRepository(GradleDsl.KOTLIN)}
+
+        dependencies {
+            // production code requires commons-lang3 at runtime, which will leak into tests' runtime classpaths
+            implementation("org.apache.commons:commons-lang3:3.11")
+        }
+
+        testing {
+            suites {
+                val test by getting(JvmTestSuite::class) {
+                    dependencies {
+                        implementation(mapOf("group" to "com.google.guava", "name" to "guava", "version" to "30.1.1-jre"))
```

Comment:
We expect these to break with future changes, yes? Might be worth adding a comment somewhere.

Reply:
I'll note this next to the map usage we expect to break.